compiler/codegen/expr.cpp

```diff
@@ -3088,6 +3088,43 @@ GenRet codegenCastToVoidStar(GenRet value)
   return ret;
 }
 
+void codegenCallPrintf(const char* arg) {
+  GenInfo* info = gGenInfo;
+
+  if (info->cfile) {
+    fprintf(info->cfile, "printf(\"%s\");", arg);
+  } else {
+#ifdef HAVE_LLVM
+    codegenCallPrintf(arg, "", "");
```

Comment:
I think it makes more sense to inline the codegenCall here. In any case, it should call the same thing with printf as the C version. As it is now, it's not using the `%s` as the first argument.

Alternatives are to to make a version of codegenCallPrintf accepting a std::vector or to use `astr` to do the concatenation and just make one accepting a single argument.

Of those, my personal preference would be an `astr` strategy. In that strategy, only the 1-arg version of codegenCallPrintf exists, and it always calls printf with two arguments - `"%s"` and then the passed argument. At the call site, you'd change e.g.

```
codegenCallPrintf("  %s: %s\\n", env->first.c_str(), env->second); 
```

into

```
codegenCallPrintf(astr(env->first.c_str(), ": ", env->second));
```

In any case please make sure to test with both C and LLVM code generation before you ask for the next review.